src/compiler/types.ts

```diff
@@ -36,6 +36,22 @@ namespace ts {
         push(...values: T[]): void;
     }
 
+    /* @internal */
+    export type EqualityComparer<T> = (a: T, b: T) => boolean;
+
+    /* @internal */
+    export type Comparer<T> = (a: T, b: T) => Comparison;
+
+    /* @internal */
+    export const enum Comparison {
+        LessThan    = -1,
+        EqualTo     = 0,
+        GreaterThan = 1
+    }
+
+    /* @internal */
+    export type Selector<T, U> = (v: T) => U;

```

Comment:
I'm not so hot on this, this is basically just a way to avoid writing an argument type?